libtenzir/builtins/operators/buffer.cpp

```diff
@@ -0,0 +1,408 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/detail/enum.hpp>
+#include <tenzir/detail/weak_run_delayed.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/tql2/plugin.hpp>
+#include <tenzir/uuid.hpp>
+
+#include <caf/typed_event_based_actor.hpp>
+
+#include <queue>
+
+namespace tenzir::plugins::buffer {
+
+TENZIR_ENUM(buffer_policy, block, drop);
```

Comment:
I had to move this outside of the anonymous namespace as gcc shows a bogus unused function warning. Something to perhaps investigate at a later point in time.

Reply:
I left a comment in 28ae3c2e65d